src/libraries/System.Runtime.InteropServices.JavaScript/tests/System.Runtime.InteropServices.JavaScript.UnitTests/System/Runtime/InteropServices/JavaScript/WebWorkerTestHelper.cs

```diff
@@ -171,14 +141,21 @@ public enum ExecutorType
         NewThread,
         JSWebWorker,
     }
+
     public class Executor
     {
         public int ExecutorTID;
+        public SynchronizationContext ExecutorSynchronizationContext;
+        public static SynchronizationContext MainSynchronizationContext;
 
         public ExecutorType Type;
 
         public Executor(ExecutorType type)
         {
+            if (MainSynchronizationContext == null)
+            {
+                MainSynchronizationContext = GetMainSynchronizationContext();
```

Comment:
i feel like MainSynchronizationContext should be a property getter that auto-initializes itself instead of working like this, but it's probably not an issue since this is used in tests

Reply:
I'm undecided if it should be public API